--- scripts/Makefile.am
+++ scripts/Makefile.am
@@ -31,7 +31,7 @@ if CROSSCOMPILING
 	@echo "NOTICE: we can't generate zsh completion when cross-compiling!"
 else # if not cross-compiling:
 	@if ! test -x "$(PERL)"; then echo "No perl: can't install zsh.pl"; exit 0; fi
-	$(PERL) $(srcdir)/zsh.pl $(top_builddir)/src/curl$(EXEEXT) > $@
+	LD_LIBRARY_PATH=$(top_builddir)/lib/.libs $(PERL) $(srcdir)/zsh.pl $(top_builddir)/src/curl$(EXEEXT) > $@
 endif
 
 install-data-local:
--- scripts/Makefile.in
+++ scripts/Makefile.in
@@ -547,7 +547,7 @@ all-local: $(ZSH_COMPLETION_FUNCTION_FIL
 $(ZSH_COMPLETION_FUNCTION_FILENAME): zsh.pl
 @CROSSCOMPILING_TRUE@	@echo "NOTICE: we can't generate zsh completion when cross-compiling!"
 @CROSSCOMPILING_FALSE@	@if ! test -x "$(PERL)"; then echo "No perl: can't install zsh.pl"; exit 0; fi
-@CROSSCOMPILING_FALSE@	$(PERL) $(srcdir)/zsh.pl $(top_builddir)/src/curl$(EXEEXT) > $@
+@CROSSCOMPILING_FALSE@	LD_LIBRARY_PATH=$(top_builddir)/lib/.libs $(PERL) $(srcdir)/zsh.pl $(top_builddir)/src/curl$(EXEEXT) > $@
 
 install-data-local:
 @CROSSCOMPILING_TRUE@	@echo "NOTICE: we can't install zsh completion when cross-compiling!"
